src/main/java/net/minecraftforge/datafix/ForgeDataFixerDelegateHandler.java

```diff
@@ -0,0 +1,194 @@
+
+        //All schemas are already reset via the mod event bus as this time.
+        //Rebuild of those is not needed.
+        this.activeFixer = new InnerFixer(new Int2ObjectAVLTreeMap<Schema>(schemas), new ArrayList<>(globalList), new IntAVLTreeSet(fixerVersions));
+
+        //This section was added to make it easier to debug the rule set at runtime, since it is now not pre-computed off thread anymore,
+        //But only on the main thread, at the moment it is needed.
+        //This basically disables the REWRITE_CACHE prefill and fills it on-demand.
+        //Although it is mainly used for debugging, it also provides a good way of reducing DFU's memory footprint, when it is not needed.
+        //Since the following code will compute an update rule from every version to the current for every known type in the schema,
+        //relating to this version. By default, however this is disabled to preserve vanilla compatibility, since it does do the pre-compute
+        //to fill up the cache.
+        //To enable it start the JVM with the system property set to true, -Dforge.datafixer.disablePreCompute=true needs to be added as
+        //a JVM Launch argument.
+        if (System.getProperty("forge.datafixer.disablePreCompute", "false").trim().toLowerCase(Locale.ROOT).equals("true")) {
+            LOGGER.warn("The pre-compute of the data fixer rules is disabled, this is not recommended for normal use, but can be useful for debugging or in low-memory situations.");
+            return;
+        }
+
+        //Get an iterator for all known versions.
+        final IntBidirectionalIterator iterator = activeFixer.fixerVersions().iterator();
+        while (iterator.hasNext())
+        {
+            //Grab the key.
+            final int versionKey = iterator.nextInt();
+
+            //Grab its schema.
+            final Schema schema = schemas.get(versionKey);
+
+            //For each type in the schema, compute its rewrite rule for the current version.
+            for (final String typeName : schema.types())
+            {
+                //Schedule the recompute off-thread
+                CompletableFuture.runAsync(() -> {
+                    //Grab the type.
+                    final Type<?> dataType = schema.getType(() -> typeName);
+
+                    //Grab its normal rewrite rule for the given version and our current target version.
+                    final TypeRewriteRule rule = activeFixer.getRule(DataFixUtils.getVersion(versionKey), dataVersion);
+
+                    //Optimize the rule to only apply to the given type.
+                    dataType.rewrite(rule, DataFixerUpper.OPTIMIZATION_RULE);
+                }, executor).exceptionally(e -> {
+                    //Fails.
+                    LOGGER.error("Unable to build datafixers", e);
+
+                    //Just kill it, if this fails we have a problem anyway!
+                    Runtime.getRuntime().exit(1);
```

Comment:
This seems a bit extreme, to quit the application when datafixers cannot be built. Is there no other more user-friendly way to bubble up the exception?

Reply:
This is literally what vanilla MC does.....

Since I do not want to change that behavior and DFU will in that case later on crash with a weird exception, I prefer the log entry and then just stop the engine from there, besides this is run in an executor which does not really allow us to bubble up the exception neatly, in this case the game just stops with the DFU error log line as the last line with the exception as well, which seems to be the cleanest option in my opinion.